packages/libs/hd-wallet/src/index.ts

```diff
@@ -1 +1,257 @@
 export * from './utils';
+import type { IUnsignedCommand } from '@kadena/client';
+import { verifySig } from '@kadena/cryptography-utils';
+import * as bip39 from '@scure/bip39';
+import { wordlist } from '@scure/bip39/wordlists/english';
+import {
+  decryptPrivateKey,
+  encryptOrEncodeSeed,
+  encryptPrivateKey,
+  extractSeedBuffer,
+} from './utils/encrypt';
+import { deriveKeyPair, signWithKeyPair, signWithSeed } from './utils/sign';
+
+/**
+ * Changes the password of an encrypted private key.
+ *
+ * @param {string} privateKey - The encrypted private key as a Base64 encoded string.
+ * @param {string} oldPassword - The current password used to encrypt the private key.
+ * @param {string} newPassword - The new password to encrypt the private key with.
+ * @returns {string} - The newly encrypted private key as a Base64 encoded string.
+ * @throws {Error} - Throws an error if the old password is empty, new password is incorrect empty passwords are empty, or if encryption with the new password fails.
+ */
+export function kadenaChangePassword(
+  privateKey: string,
+  oldPassword: string,
+  newPassword: string,
+): string {
+  if (oldPassword === '' || oldPassword === undefined) {
+    throw new Error('The old password cannot be empty.');
+  }
+  if (newPassword === '' || newPassword === undefined) {
+    throw new Error('The new password cannot be empty.');
+  }
+  if (oldPassword === newPassword) {
+    throw new Error(
+      'The new password must be different from the old password.',
+    );
+  }
+
+  let decryptedPrivateKey;
+  try {
+    decryptedPrivateKey = decryptPrivateKey(privateKey, oldPassword);
+  } catch (error) {
+    throw new Error(
+      `Failed to decrypt the private key with the old password: ${error.message}`,
+    );
+  }
+
+  let newEncryptedPrivateKey;
+  try {
+    newEncryptedPrivateKey = encryptPrivateKey(
+      decryptedPrivateKey,
+      newPassword,
+    );
+  } catch (error) {
+    throw new Error(
+      `Failed to encrypt the private key with the new password: ${error.message}`,
+    );
+  }
+
+  return newEncryptedPrivateKey;
+}
+
+/**
+ * Convert a given mnemonic phrase into a seed buffer.
+ *
+ * @param {string} mnemonic - A mnemonic seed phrase to be converted into a seed buffer.
+ * @param {string} [password] - Optional password for encrypting the seed.
+ * @throws {Error} Throws an error if the provided mnemonic is not valid.
+ * @returns {Promise<{ seedBuffer: Uint8Array, seed: string }>} - Returns the seed buffer and processed seed.
+ */
+export async function kadenaGenSeedFromMnemonic(
+  mnemonic: string,
+  password?: string,
+): Promise<{ seedBuffer: Uint8Array; seed: string }> {
+  if (bip39.validateMnemonic(mnemonic, wordlist) === false) {
+    throw Error('Invalid mnemonic.');
+  }
+
+  const seedBuffer = await bip39.mnemonicToSeed(mnemonic);
+  let seed: string;
+
+  if (typeof password === 'string' && password.length > 0) {
```

Comment:
why don't we allow empty string as password?

Reply:
why would we? (please elaborate a case where it would be usefull?)

Reply:
From a wallet perspective, we're planning on allowing multiple mnemonics. For testnet and devnet I wouldn't care whether it has a password, so it being empty could be fine